design/src/rfcs/rfc0012_crates_io_publishing.md

```diff
@@ -0,0 +1,214 @@
+- `patch`:
+    - Bug fixes that do not break backwards compatibility
+    - Model updates that _only_ have documentation changes
+
+During phase 3, bumps to the `major` version must be coordinated across all SDK and runtime crates.
+
+### Release Identification
+
+Since there will no longer be one SDK "version", release tags will be dates in `YYYY-MM-DD` format
+rather than version numbers. Additionally, the SDK's user agent string will need to include a separate
+service version number (this requirement has already been implemented).
+
+### Yanking
+
+It must be possible to yank an entire release with a single command-line. The publisher tool must
+be updated to understand which crate versions were released with a given release tag, and be able to
+yank that entire tag worth of crates.
+
+Phase 1: New Versioning Strategy
+--------------------------------
+
+Phase 1 will address the following challenges introduced by uncoordinating the `major.minor` versions:
+- Tracking of versions associated with a release tag
+- Creation of version bump process for code generated crates
+- Enforcement of version bump process in runtime crates
+- Yanking of versions associated with a release tag
+
+### Version Tracking
+
+A new manifest file will be introduced in the root of [aws-sdk-rust] named `versions.toml` that describes
+all versioning information for any given commit in the repository. In the main branch, the `versions.toml`
+in tagged commits will become the source of truth for which crate versions belong to that release, as well
+as additional metadata that's required for maintaining version process in the future.
+
+This format will look as follows:
+```toml
+smithy_rs_version = "<release-tag|commit-hash>"
+
+[aws-smithy-types]
+version = "0.50.1"
+
+[aws-config]
+version = "0.40.0"
+
+[aws-sdk-s3]
+version = "0.89.0"
+model_hash = "<hash>"
+
+# ...
+```
+
+The auto-sync tool is responsible for maintaining this file. When it generates a new SDK, it will take
+the version numbers from runtime crates directly, and it will use the rules from the next section to determine
+the version numbers for the generated crates.
+
+### Versioning for Code Generated Crates
+
+Code generated crates will have their `minor` version bumped when the version of [smithy-rs] used to generate
+them changes, or when model updates with API changes are made. Three pieces of information are required to
+handle this process: the previously released version number, the [smithy-rs] version used to generate the code,
+and the level of model updates being applied. For this last one, if there are multiple model updates that
+affect only documentation, but then one model update that affects an API, then as a whole they will be
+considered as affecting an API and require a `minor` version bump.
+
+The previously released version number will be retrieved from crates.io using its API. The [smithy-rs] version
```

Comment:
conceivably we could track the previously deployed version in versions.toml right?